Snake/src/snakeBoard.js

```diff
@@ -0,0 +1,446 @@
+            this.verticalAdjustment = 0;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyUp && lastKeyPressed !== keyDown) {
+            this.horizontalAdjustment = 0;
+            this.verticalAdjustment = -this.squareSize;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyRight && lastKeyPressed !== keyLeft) {
+            this.horizontalAdjustment = this.squareSize;
+            this.verticalAdjustment = 0;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+
+        if (keyPressed === keyDown && lastKeyPressed !== keyUp) {
+            this.horizontalAdjustment = 0;
+            this.verticalAdjustment = this.squareSize;
+
+            this.setLastKeyPressed(keyPressed);
+        }
+    }
+
+    getRandomPosition(board) {
+        const x = Math.round(Math.random() * board.current.width);
+        const y = Math.round(Math.random() * board.current.height);
+
+        return {
+            x: x - x % this.squareSize + (this.squareSize / 2),
+            y: y - y % this.squareSize + (this.squareSize / 2)
+        };
+    }
+
+    setFoodPosition(foodPos) {
+        this.setState({ food: foodPos });
+    }
+
+    drawFood() {
+        const snakeBoardStyles = this.getSnakeBoardStyles();
+        const radius = this.props.snakeSize / 2;
+        let { x: foodX, y: foodY } = this.state.food;
+
+        snakeBoardStyles.beginPath();
+        snakeBoardStyles.arc(foodX, foodY, radius, 0, 2 * Math.PI, false);
+        snakeBoardStyles.fillStyle = 'lightgreen';
+        snakeBoardStyles.fill();
+        snakeBoardStyles.strokeStyle = 'green';
+        snakeBoardStyles.stroke();
+    }
+
+    enlargeIfFoodEaten() {
+        const head = { x: this.state.snake[0].x + (this.props.snakeSize / 2), y: this.state.snake[0].y + (this.props.snakeSize / 2) };
+
+        if (head.x === this.state.food.x && head.y === this.state.food.y) {
+            this.setFoodPosition(this.getRandomPosition(this.boardRef));
+            this.setState({ snake: this.enLarge(head, this.state.snake) });
+
+            this.pieceCounter++;
+            this.score += 10;
+        }
+    }
+
+    enLarge(head, snake) {
+        const newSnake = [...snake];
+
+        newSnake.push([{ x: head - this.pieceCounter * this.horizontalAdjustment, y: head - this.pieceCounter * this.verticalAdjustment }]);
+
+        return newSnake
+    }
+
+    checkForCollision() {
+        const head = { x: this.state.snake[0].x, y: this.state.snake[0].y };
+
+        for (let i = 1; i < this.state.snake.length; i++) {
+            if (head.x === this.state.snake[i].x && head.y === this.state.snake[i].y) return true;
+        }
+    }
+
+    checkForCollisionWithBorders(board) {
+        let newSnakePos = [...this.state.snake];
+        const head = { x: this.state.snake[0].x, y: this.state.snake[0].y };
+        const snakeBoard = board.current;
+        const leftWallCollision = newSnakePos[0].x < 0;
```

Comment:
Isnt height and width the same size as well the start on both orientations are 0